add flag serial command

GitMeGet opened the pull request:
- updated TS_Settings to include upload_flag

GitMeGet committed:
add upload_flag to struct TS_Settings

GitMeGet committed:
add serial cmd: flag

jamesi8086 commented on alpha/serial_cmd.cpp:
```diff
@@ -209,6 +217,72 @@ static int do_clock_cmd(int argc, char **argv)
+        printf("Password exceeds max string length of %d\n\n", STR_ARG_MAXLEN);
+      }
+    }
+    TS_Storage.settings_save();
```
Oh wait, can you make save a separate command?

We need to be able to change settings w/o saving to flash. This is particularly useful during automated testing.

GitMeGet replied:
Got it. So perhaps default cmd saves to EEPROM and --ram saves to RAM

yongkimleng replied:
ah no, just have an explicit command that saves

we actually have a settings_load to overwrite whatever is in RAM w/ whats in flash

jamesi8086 replied:
your method works

jamesi8086 commented on alpha/serial_cmd.cpp:
```diff
@@ -226,6 +300,95 @@ static void register_clock_cmd()
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    TS_Storage.settings_save();
```
same

jamesi8086 reviewed (approved):
note on the save command

GitMeGet committed:
add flag to save to RAM or EEPROM

- for wifi and flag cmd

GitMeGet committed:
Merge branch 'master' into yp-flag-cmd

GitMeGet committed:
organise code

- place related functions together:
e.g. flag_args, do_flag_cmd(), register_flag_cmd()

GitMeGet committed:
wifi cmd: add flag to save to RAM or EEPROM

GitMeGet committed:
modify clock cmd:

- print clock help if no args provided
- add clock -s help message example

GitMeGet committed:
refactor wifi cmd to use check_copy_str_setting()

GitMeGet commented:
added RAM flag

jamesi8086 commented on alpha/serial_cmd.cpp:
```diff
@@ -450,8 +451,7 @@ static int do_userid_cmd(int argc, char **argv)
     check_copy_str_setting(userid_args.set->sval[0], settings->userId);
-    TS_Storage.settings_save();
-    printf("Setting saved\n\n");
+    save_to_RAM_or_EEPROM(userid_args.ram_flag);
```
no mixed cases in function name

jamesi8086 commented on alpha/serial_cmd.cpp:
```diff
@@ -241,8 +241,7 @@ static int do_clock_cmd(int argc, char **argv)
   {
-    printf("Type help for help\n\n");
-    return ESP_ERR_INVALID_ARG;
+    print_cmd_help(argv[0], (void**) &clock_args);
```
clockArgs

jamesi8086 commented:
I probably should have written a style guide:

filenames: snake_case_filenames
function names: snake_case
variable names: camelCase
type names: PascalCase with the exception of adding prefixes, eg. TS_ or _
defines: UPPER_CASE

GitMeGet committed:
fix code-style

GitMeGet commented:
If using platformio, reduce the debug_level in platformio.ini to make it easier for your eyes :)

jamesi8086 commented:
can merge if no conflicts
